crates/core/component/dex/src/component/router/route_and_fill.rs

```diff
@@ -55,10 +52,20 @@ pub trait HandleBatchSwaps: StateWrite + Sized {
         } else {
             // There was no input for asset 1, so there's 0 output for asset 2 from this side.
             tracing::debug!("no input for asset 1, skipping 1=>2 execution");
-            (0u64.into(), delta_1)
+            SwapExecution {
+                traces: vec![],
+                input: Value {
+                    asset_id: trading_pair.asset_1(),
+                    amount: Amount::zero(),
+                },
+                output: Value {
+                    asset_id: trading_pair.asset_2(),
+                    amount: Amount::zero(),
+                },
+            }
```

Comment:
Since we skipped execution, shouldn't we not write a `SwapExecution` into the state at all, rather than making an empty one? Otherwise a caller has to differentiate between "swap execution that is present and happened" and "swap execution that is present but did not happen" and "swap execution that is not present and did not happen".